### ISSUE TITLE
Replace memory-model with vm-memory from rust-vmm

### ISSUE THREAD

**Issue by andreeaflorescu** [firecracker-microvm/firecracker#1047]
This task is currently blocked by vm-memory not being published on crates.io.
Check the repository here: https://github.com/rust-vmm/vm-memory

@bonzini did the switch in his [fork](https://github.com/bonzini/firecracker/commit/5e84f75d599995462fcc1765ed511846f36fedc0), but it's not ready for merging in Firecracker as per his [comment](https://github.com/rust-vmm/vm-memory/pull/9#issuecomment-480050928).

**Comment by serban300** [comment]
This seems a duplicate of #1137 . I think we can close it.
